### PR TITLE
Created a harmony-basics tutorial 

### PR DESCRIPTION
**Pull request by soggylithe**
Created a tutorial with the basics of Harmony consolidated into one place. Should be enough to get someone up and running with no previous knowledge of Harmony or little coding knowledge.

### CODE REVIEW

File: modding-tutorials/harmony-basics.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+        harmonyInstance.UnpatchAll();
+        Destroy(gameObject);
+        Debug.Log("Mod Brine has been unloaded!");
+    }
+```
+
+In older versions you needed to `Destroy()` the mod object when unloading but that is also handled gracefully these days.
+
+## Traversing
```
Comment (tr808axm) [inline modding-tutorials/harmony-basics.md:181]: It looks like something is missing here. Please either delete the heading or add a paragraph.

Comment (tr808axm) [inline modding-tutorials/harmony-basics.md:181]: If you want, you can add your name at the bottom of the tutorial, but you don't need to :)

File: modding-tutorials/harmony-basics.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+---
+###### [HarmonyTranspiler]
+Transpilers are a much more advanced way to get your own code to run. It is a bit out of the scope of a tutorial here so I will link to [this article](https://harmony.pardeike.net/articles/patching-transpiler.html#basic-transpiler-tutorial)
+
+To give a simplified overview: Instead of writing code to be executed, you are looking at all of the IL instructions that a chunk of code represents and doing some of your own instructions whenever you think the right time is. 
+
+---
+###### [HarmonyTargetMethod]
+I'll be honest I don't understand this one and works other than it is a different way to tell Harmony which method you are changing instead of using the class attribute
```
Comment (tr808axm) [inline modding-tutorials/harmony-basics.md:113]: `[HarmonyTargetMethod]` can be used to dynamically choose a target method. If you can't explain this one, leave it out of the docs.

**Review by tr808axm — CHANGES_REQUESTED** [review]

**Comment by tr808axm** [comment]
You'll also need to add a link to your tutorial to the `SUMMARY.md` file at the root of this repo. The summary is basically a table of contents for the docs.

File: modding-tutorials/harmony-basics.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+[HarmonyPatch(typeof(BlockCreator), "CanBuildBlock")]
+public class HarmonyPatch_IgnoreCollisionOnAlt
+{
+	[HarmonyPrepare]
+	static bool IsCoolModInstalled()
+	{...}
+	
+    [HarmonyPostfix]
+    static BuildError CheckForAlt(BuildError __result)
+    {...}
```
[the remark this answers is not in the document]
Comment (soggylithe) [reply modding-tutorials/harmony-basics.md:22]: Do you mean just opening the curly brackets with nothing in them? The indents currently match up with all the other code blocks. Classes are flush, methods are 1tab and nested brackets are inline with contents tabbed up one.

Would this work better:
    [HarmonyPostfix]
    static BuildError CheckForAlt(BuildError __result)
    {
    }

or:
    [HarmonyPostfix]
    static BuildError CheckForAlt(BuildError __result)
    {
        /*Code here*/
    }

[the remark this answers is not in the document]
Comment (tr808axm) [reply modding-tutorials/harmony-basics.md:22]: Your braces style is fine. What I mean is the fact that the `CheckForAlt` method is indented with spaces and the rest is indented with tabs. Here's a screenshot that demonstrates my point:

image

**Review by tr808axm — CHANGES_REQUESTED** [review]

**Review by tr808axm — APPROVED** [review]
Looks good!

**Comment by tr808axm** [comment]
Thanks a lot for your contribution!

**Pull request merged**